Result/ResultType.swift

```diff
@@ -73,10 +73,10 @@ public extension ResultType {
 /// Protocol used to constrain `tryMap` to `Result`s with compatible `Error`s.
 public protocol ErrorTypeConvertible: ResultErrorType {
 	associatedtype ConvertibleType = Self
-	static func errorFromErrorType(error: ResultErrorType) -> Self
+	static func errorFromErrorType(error: ResultErrorType) -> ConvertibleType
 }
 
-public extension ResultType where Error: ErrorTypeConvertible {
+public extension ResultType where Error: ErrorTypeConvertible, Error.ConvertibleType == Error {
```

Comment:
I added the constraint to make [the implementation here](https://github.com/antitypical/Result/pull/147/files#diff-77b13e3ea6c630c28c5afd2ccbd7904dR88) safer.